moveit_core/collision_detection/include/moveit/collision_detection/world_diff.h

```diff
@@ -59,8 +59,9 @@ class WorldDiff
   ~WorldDiff();
 
   /** \brief Set which world to record.  Records all objects in old world (if
-   * any) as DESTROYED and all objects in new world as CREATED and ADD_SHAPE
-   * */
+   * any) as DESTROYED and all objects in new world as CREATED and ADD_SHAPE.
+   * Warning: function output depends on the class member `world_`. This makes it an "impure function"
```

Comment:
This is the setter for world_. What other side effects do you expect to happen here?